### ISSUE TITLE
Improve determination of cause of an explosion

### ISSUE THREAD

**Issue by MegadreamsBE** [multitheftauto/mtasa-blue#681]
**Is your feature request related to a problem? Please describe.**
Explosions cannot only be caused by a player/vehicle/object but also by a projectile. Sure, this projectile is likely created by one of the former specified elements but that information is already stored in the projectile itself. In the end the explosion is really caused by that projectile.

**Describe the solution you'd like**
The "onClientExplosion" event should also recognize projectiles as a possible cause for explosions and do a better overall job at determining the actual cause of an explosion.

**Describe alternatives you've considered**
Well you could track the amount of time left for a projectile to explode and if it's about to hit a ped/object/vehicle but that's inaccurate.

**Comment by qaisjp** [comment]
I agree it would be nice to have this context. Pull requests here would be greatly appreciated.
